bitcoin/BRPeer.c

```diff
@@ -909,18 +912,18 @@ static int _BRPeerOpenSocket(BRPeer *peer, int domain, double timeout, int *erro
             addrLen = sizeof(struct sockaddr_in);
         }
         
-        if (connect(ctx->socket, (struct sockaddr *)&addr, addrLen) < 0) err = errno;
+        if (connect(sock, (struct sockaddr *)&addr, addrLen) < 0) err = errno;
         
         if (err == EINPROGRESS) {
             err = 0;
             optLen = sizeof(err);
             tv.tv_sec = timeout;
             tv.tv_usec = (long)(timeout*1000000) % 1000000;
             FD_ZERO(&fds);
-            FD_SET(ctx->socket, &fds);
-            count = select(ctx->socket + 1, NULL, &fds, NULL, &tv);
+            FD_SET(sock, &fds);
+            count = select(sock + 1, NULL, &fds, NULL, &tv);
```

Comment:
Just noticed this; its been a while since I've written Berkeley socket code but how is the first argument the socket file descriptor?